ddtrace/tracer/option.go

```diff
@@ -187,15 +188,22 @@ const maxPropagatedTagsLength = 512
 func newConfig(opts ...StartOption) *config {
 	c := new(config)
 	c.sampler = NewAllSampler()
-	c.agentURL = "http://" + resolveAgentAddr()
-	c.httpClient = defaultHTTPClient()
-	if url := internal.AgentURLFromEnv(); url != nil {
-		if url.Scheme == "unix" {
-			c.httpClient = udsClient(url.Path)
-		} else {
-			c.agentURL = url.String()
+	if c.agentURL == nil {
```

Comment:
Won't `c.agentURL` always be nil here since the `config` has just been created?

Reply:
Yes, this was actually a problem causing the issue @mackjmr noticed. I've moved it down.